lib/jsdom/living/nodes/HTMLTableCellElement-impl.js

```diff
@@ -3,50 +3,63 @@
 const HTMLElementImpl = require("./HTMLElement-impl").implementation;
 
 const closest = require("../helpers/traversal").closest;
-const createDOMTokenList = require("../dom-token-list").create;
-const resetDOMTokenList = require("../dom-token-list").reset;
 
-class HTMLTableCellImpl extends HTMLElementImpl {
-  get headers() {
-    if (this._headers === undefined) {
-      this._headers = createDOMTokenList(this, "headers");
-    }
-    return this._headers;
+function reflectedAttributeClampedToRange(attrValue, min, max, defaultValue = 0) {
+  if (attrValue === null) {
+    return defaultValue;
   }
-
-  get cellIndex() {
-    const tr = closest(this, "tr");
-    if (tr === null) {
-      return -1;
-    }
-
-    return tr.cells.indexOf(this);
+  // We don't allow hexadecimal numbers here.
+  // eslint-disable-next-line radix
+  const parsed = parseInt(attrValue, 10);
+  if (isNaN(parsed) || parsed < 0) {
+    return defaultValue;
   }
+  if (parsed < min) {
+    return min;
+  }
+  if (parsed > max) {
+    return max;
+  }
+  return parsed;
+}
 
+class HTMLTableCellImpl extends HTMLElementImpl {
   get colSpan() {
-    const value = this.getAttribute("colspan");
-    return value === null ? 1 : value;
+    return reflectedAttributeClampedToRange(this.getAttribute("colspan"), 1, 1000, 1);
   }
 
   set colSpan(V) {
     this.setAttribute("colspan", String(V));
   }
 
   get rowSpan() {
-    const value = this.getAttribute("rowspan");
-    return value === null ? 1 : value;
+    return reflectedAttributeClampedToRange(this.getAttribute("rowspan"), 0, 65534, 1);
   }
 
   set rowSpan(V) {
     this.setAttribute("rowspan", String(V));
   }
 
-  _attrModified(name, value, oldValue) {
-    if (name === "headers" && this._headers) {
-      resetDOMTokenList(this._headers, value);
+  get cellIndex() {
+    const tr = closest(this, "tr");
+    if (tr === null) {
+      return -1;
+    }
+
+    return tr.cells.indexOf(this);
+  }
+
+  get scope() {
+    const value = this.getAttribute("scope");
+    if (value === "row" || value === "col" || value === "rowgroup" || value === "colgroup") {
```

Comment:
This should match [ascii case insensitive](https://html.spec.whatwg.org/#enumerated-attribute).